// databox/dist/models/ModelApiResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Static OpenAPI document of Push API resource
 * Push API resources Open API documentation
 *
 * The version of the OpenAPI document: 0.4.3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfModelApiResponse = instanceOfModelApiResponse;
exports.ModelApiResponseFromJSON = ModelApiResponseFromJSON;
exports.ModelApiResponseFromJSONTyped = ModelApiResponseFromJSONTyped;
exports.ModelApiResponseToJSON = ModelApiResponseToJSON;
/**
 * Check if a given object implements the ModelApiResponse interface.
 */
function instanceOfModelApiResponse(value) {
    return true;
}
function ModelApiResponseFromJSON(json) {
    return ModelApiResponseFromJSONTyped(json, false);
}
function ModelApiResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'status': json['status'] == null ? undefined : json['status'],
        'message': json['message'] == null ? undefined : json['message'],
    };
}
function ModelApiResponseToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'status': value['status'],
        'message': value['message'],
    };
}
