// databox/dist/models/PushData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Static OpenAPI document of Push API resource
 * Push API resources Open API documentation
 *
 * The version of the OpenAPI document: 0.4.3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPushData = instanceOfPushData;
exports.PushDataFromJSON = PushDataFromJSON;
exports.PushDataFromJSONTyped = PushDataFromJSONTyped;
exports.PushDataToJSON = PushDataToJSON;
var PushDataAttribute_1 = require("./PushDataAttribute");
/**
 * Check if a given object implements the PushData interface.
 */
function instanceOfPushData(value) {
    return true;
}
function PushDataFromJSON(json) {
    return PushDataFromJSONTyped(json, false);
}
function PushDataFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'attributes': json['attributes'] == null ? undefined : (json['attributes'].map(PushDataAttribute_1.PushDataAttributeFromJSON)),
        'date': json['date'] == null ? undefined : json['date'],
        'key': json['key'] == null ? undefined : json['key'],
        'periodFrom': json['periodFrom'] == null ? undefined : json['periodFrom'],
        'periodTo': json['periodTo'] == null ? undefined : json['periodTo'],
        'unit': json['unit'] == null ? undefined : json['unit'],
        'value': json['value'] == null ? undefined : json['value'],
    };
}
function PushDataToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'attributes': value['attributes'] == null ? undefined : (value['attributes'].map(PushDataAttribute_1.PushDataAttributeToJSON)),
        'date': value['date'],
        'key': value['key'],
        'periodFrom': value['periodFrom'],
        'periodTo': value['periodTo'],
        'unit': value['unit'],
        'value': value['value'],
    };
}
