// databox/dist/models/PushDataAttribute.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Static OpenAPI document of Push API resource
 * Push API resources Open API documentation
 *
 * The version of the OpenAPI document: 0.4.3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPushDataAttribute = instanceOfPushDataAttribute;
exports.PushDataAttributeFromJSON = PushDataAttributeFromJSON;
exports.PushDataAttributeFromJSONTyped = PushDataAttributeFromJSONTyped;
exports.PushDataAttributeToJSON = PushDataAttributeToJSON;
/**
 * Check if a given object implements the PushDataAttribute interface.
 */
function instanceOfPushDataAttribute(value) {
    return true;
}
function PushDataAttributeFromJSON(json) {
    return PushDataAttributeFromJSONTyped(json, false);
}
function PushDataAttributeFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'key': json['key'] == null ? undefined : json['key'],
        'value': json['value'] == null ? undefined : json['value'],
    };
}
function PushDataAttributeToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'key': value['key'],
        'value': value['value'],
    };
}
