// databox/dist/models/State.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Static OpenAPI document of Push API resource
 * Push API resources Open API documentation
 *
 * The version of the OpenAPI document: 0.4.3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.State = void 0;
exports.instanceOfState = instanceOfState;
exports.StateFromJSON = StateFromJSON;
exports.StateFromJSONTyped = StateFromJSONTyped;
exports.StateToJSON = StateToJSON;
/**
 *
 * @export
 */
exports.State = {
    Down: 'DOWN',
    Up: 'UP'
};
function instanceOfState(value) {
    for (var key in exports.State) {
        if (Object.prototype.hasOwnProperty.call(exports.State, key)) {
            if (exports.State[key] === value) {
                return true;
            }
        }
    }
    return false;
}
function StateFromJSON(json) {
    return StateFromJSONTyped(json, false);
}
function StateFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
function StateToJSON(value) {
    return value;
}
